src/bytes.rs

```diff
@@ -3,6 +3,7 @@ use core::{
     fmt::Debug,
     hash::Hash,
     ops::{Deref, RangeBounds},
+    ptr::NonNull,
```

Comment:
Style nit (which I've tried to follow in the rest of the crate): `NonNull`, alone, does not really specify _what_ is non-zero, even if it is a very pervasive stdlib type).

I personally thus prefer to use `ptr::NonNull` where needed 🙂

```suggestion
    ptr,
```

  - (I also like importing `mem` to get access to the very often needed `mem::size_of::<_>()` operations and whatnot.)